waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/exchange/ExchangeExternalSignaturesController.kt

```diff
@@ -355,6 +350,176 @@ fun Application.exchangeExternalSignatures() = walletRoute {
+                )
+                context.respond(
+                    HttpStatusCode.OK,
+                    responsePayload,
+                )
+            }.onFailure { error ->
+                context.respond(
+                    HttpStatusCode.BadRequest,
+                    error.message ?: "Unknown error",
+                )
+            }
+        }
+
+        post("external_signatures/offer/submit", {
+            summary = "Submission (second) step for an OID4VCI flow with externally provided signatures."
+
+            request {
+                body<SubmitOID4VCIRequest> {
+                    required = true
+                    example("When proofType == cwt") {
+                        value = SubmitOID4VCIRequest(
+                            did = "did:web:walt.id",
+                            offerURL = "openid-credential-offer://?credential_offer=",
+                            offeredCredentialProofsOfPossession = listOf(
+                                IssuanceService.OfferedCredentialProofOfPossession(
+                                    OfferedCredential(
+                                        format = CredentialFormat.mso_mdoc,
+                                    ),
+                                    ProofType.cwt,
+                                    "<<BASE64URL-ENCODED SIGNED CWT>>",
+                                )
+                            ),
+                            credentialIssuer = "https://issuer.portal.walt.id"
+                        )
+                    }
+                    example("When proofType == jwt") {
+                        value = SubmitOID4VCIRequest(
+                            did = "did:web:walt.id",
+                            offerURL = "openid-credential-offer://?credential_offer=",
+                            offeredCredentialProofsOfPossession = listOf(
+                                IssuanceService.OfferedCredentialProofOfPossession(
+                                    OfferedCredential(
+                                        format = CredentialFormat.jwt_vc_json,
+                                    ),
+                                    ProofType.jwt,
+                                    "<<COMPACT-SERIALIZED SIGNED JWT>>",
+                                )
+                            ),
+                            credentialIssuer = "https://issuer.portal.walt.id"
+                        )
+                    }
+                }
+            }
+
+            response(OpenAPICommons.useOfferRequestEndpointResponseParams())
+        }) {
+            val walletService = getWalletService()
+            val req = call.receive<SubmitOID4VCIRequest>()
+            logger.debug { "Request: $req" }
+
+            val did = req.did ?: walletService.listDids().firstOrNull()?.did
+            ?: throw IllegalArgumentException("No DID to use supplied and no DID was found in wallet.")
+
+            runCatching {
```

Comment:
this could probably be simplified to smth. like:
```kotlin
runCatching {
    WalletServiceManager
        .externalSignatureClaimStrategy
        .submitCredentialClaim(
            tenantId = walletService.tenant,
            accountId = walletService.accountId,
            walletId = walletService.walletId,
            pending = req.requireUserInput ?: true,
            did = did,
            offerURL = req.offerURL,
            credentialIssuerURL = req.credentialIssuer,
            accessToken = req.accessToken,
            offeredCredentialProofsOfPossession = req.offeredCredentialProofsOfPossession,
        )
}.getOrElse { throw IllegalArgumentException("Unknown error") }.let {
    context.respond(HttpStatusCode.OK, it)
}
```
in case of an exception it will throw the IAE which gets handled by [StatusPages](https://github.com/walt-id/waltid-identity/blob/5ec0016b4174cb76e31fbb04343bf0843c9f0ea3/waltid-services/waltid-service-commons/src/main/kotlin/id/walt/commons/web/plugins/StatusPages.kt#L34) as a `BadRequest`